### DISCUSSION TITLE
Rollback

### DISCUSSION THREAD

**Discussion by jaredmoody** [mattbrictson/tomo#266] [Q&A] [answered]
Is there a philosophy reason why there's no rollback command or is it just not yet implemented?

Is there a way to deploy a previous commit instead of HEAD on the main branch, or is reverting the commit how you prefer to handle it with tomo?

**Comment by mattbrictson** [comment, marked as answer]
I have been a long-time user (and maintainer) of capistrano, and in all those years I never used its rollback feature. When I decided to build tomo, I wanted to focus first and foremost on features I would use. Rollback did not make the cut. I don't see myself adding a rollback command to tomo any time soon.

My philosophy is if my team merges something onto the main branch that ends up deploying a serious bug, we will follow that up with another commit that reverts it (or applies a fix) and then we deploy again. If we are disciplined about having good test coverage, using CI, and taking advantage of a staging environment before code reaches production, then such scenarios should be exceedingly rare.

In an emergency situation where a deploy needs to be urgently reverted: yes, I would deploy a previous commit by specifying the `git_ref` setting ([docs](https://tomo-deploy.com/plugins/git/#settings)), like this:

```
tomo deploy -s git_ref=PREVIOUS_SHA
```

**Reply by jaredmoody** [reply]
Great, thanks for the info and the tip about using `-s git-ref`, I missed that in the docs!
